### PR TITLE
Mobile speech and camera

### PR DESCRIPTION
**Pull request by marjasi**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,16 @@
 /.vs
 /ShopLens/ShopLensForms/bin/Debug
 /ShopLens/ShopLensForms/obj
+/Mobile/ShopLens/.vs/ShopLens/v15/Server/sqlite3
```
Comment (mrjslau) [inline .gitignore:22]: Maybe /Mobile/ShopLens/.vs and /Mobile/ShopLens/iOS/obj  is enough instead of all the specific ones?

**Review by mrjslau — APPROVED** [review]
axml files need to be aligned. App is working correctly, code looks good for now.

File: Mobile/ShopLens/Droid/CameraActivity.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            if (IsThereAnAppToTakePictures())
+            {
+                CreateDirectoryForPictures();
+                BtnTakeImg = FindViewById<Button>(Resource.Id.btntakepicture);
+                ImgView = FindViewById<ImageView>(Resource.Id.ImgTakeimg);
+                BtnTakeImg.Click += TakeAPicture;
+            }
+        }
+
+        private void CreateDirectoryForPictures()
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/CameraActivity.cs:49]: Maybe we should have a seperate class for I.O. operations, but this doesn't seem too out-of-place here either

File: Mobile/ShopLens/Droid/CameraActivity.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            }
+            // If there's a working camera on the device.
+            if (takePictureIntent.ResolveActivity(PackageManager) != null) {
+                StartActivityForResult(takePictureIntent, REQUEST_IMAGE);
+            }
+            
+        }
+        protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
+        {
+            if (requestCode == REQUEST_IMAGE && resultCode == Result.Ok)
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/CameraActivity.cs:82]: And what if the result is not OK? Do we have a way of knowing the error message or something like that?

File: Mobile/ShopLens/Droid/SpeechActivity.cs [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+using Java.Util;
+using Android.Runtime;
+
+namespace ShopLens.Droid
+{
+    [Activity(Label = "SpeechActivity")]
+    public class SpeechActivity : Activity, TextToSpeech.IOnInitListener
+    {
+        bool isRecording;
+        readonly int VOICE = 10;
```
Comment (EdvardasDlugauskas) [inline Mobile/ShopLens/Droid/SpeechActivity.cs:17]: what exactly is this? Voice "request code"? The name could be a little more descriptive I guess

**Review by EdvardasDlugauskas — APPROVED** [review]
Minor notes, but seems good 👍

**Pull request merged**